File: potd-gfg/Add all greater values to every node in a BST (14th Jan 2022)/AddGreaterValToEveryNode.java
class Solution{
    int sum=0;
    // modify the BST and return its root
    public Node modify(Node root)
    {
        //Write your code here
        addGreater(root);
        return root;
    }
    
    public void addGreater(Node root){
        if(root==null)
            return;
        addGreater(root.right);
        int data=root.data;
        root.data+=sum;
        sum+=data;
        addGreater(root.left);
        
    }
    
}

File: potd-gfg/Add all greater values to every node in a BST (14th Jan 2022)/AddGreaterValToEveryNode.py
#User function Template for python3

'''
# Tree Node
class Node:
    def __init__(self, val):
        self.right = None
        self.data = val
        self.left = None
'''

# modify the BST and return its root
def modify(root):
   
    global S 
    S = 0
    aagvtenibst(root)
    
    return root
    
def aagvtenibst(root):
    global S
    if root:
        
        # First recur on left child
        aagvtenibst(root.right)
        
        S += root.data
        root.data = S
       
        # now recur on right child
        aagvtenibst(root.left)
